test_run.sh

```diff
@@ -63,6 +63,28 @@ test_scala_library_suite() {
   action_should_fail build test_expect_failure/scala_library_suite:library_suite_dep_on_children
 }
 
+test_scala_library_expect_failure_on_missing_direct_deps() {
+  set +e
+
+  expected_message="Target '//test_expect_failure/missing_direct_deps:direct_dependency' is used but isn't explicitly declared, please add it to the deps"
+  command='bazel build test_expect_failure/missing_direct_deps:transitive_dependency_user'
+  output=$($command 2>&1)
+  if [  $? -eq 0 ]; then
+    echo "$output"
+    echo "'bazel build of scala_library with missing direct deps should have failed."
+    exit 1
+  fi
+  echo "$output"
```

Comment:
Is this echo (the first one) helpful for debugging? not against it just trying to understand